File: BasicLearning/SwiftSp/OCDelayLoad/OCDelayLoad/ViewController.m
//
//  ViewController.m
//  OCDelayLoad
//
//  Created by leven on 2017/11/7.
//  Copyright © 2017年 lee. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSLog(@"开始加载");
    
    // 2秒
    dispatch_time_t when = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2.0 * NSEC_PER_SEC));
    /**
     延时加载

     @param when: 从现在开始经过多少纳秒
     @param dispatch_get_main_queue ：由队列调度任务
     @return return value description
     */
    dispatch_after(when, dispatch_get_main_queue(), ^{
        NSLog(@"----------------");
    });
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: BasicLearning/SwiftSp/DidSetUI/DidSetUI/ViewController.swift
//
//  ViewController.swift
//  DidSetUI
//
//  Created by leven on 2017/11/6.
//  Copyright © 2017年 lee. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let p = Person()
        p.name = "小花"
        
        let label = DemoLabel(frame:CGRect(x: 20, y: 40, width: 100, height: 40))
        label.backgroundColor = #colorLiteral(red: 0.5568627715, green: 0.3529411852, blue: 0.9686274529, alpha: 1)
        view.addSubview(label)
        
        // 将模型设置给label
        label.person = p
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

